src/Kiota.Builder/Refiners/TypeScriptRefiner.cs

```diff
@@ -250,20 +258,209 @@ private static void AddDownwardsConstantsImports(CodeElement currentElement)
+        if (function is null) return;
+
+        var method = CreateSerializerMethodForComposedType(codeInterface, function, composedType);
+        var serializerFunction = new CodeFunction(method) { Name = method.Name };
+        serializerFunction.AddUsing(function.Usings.ToArray());
+
+        RemoveOldCodeFunctionAndAddNewOne(children, codeInterface, codeNamespace, function, serializerFunction);
     }
+
+    private static void ReplaceDeserializerMethodForComposedType(CodeInterface codeInterface, CodeNamespace codeNamespace, CodeComposedTypeBase composedType, List<CodeElement> children)
+    {
+        var deserializerMethod = FindFunctionOfKind(children, CodeMethodKind.Deserializer);
+
+        if (deserializerMethod is null) return;
+
+        // For code union Deserializer is not required, however its needed for Object Intersection types
+        if (composedType is CodeIntersectionType && !IsComposedOfPrimitives(composedType))
+        {
+            var method = CreateDeserializerMethodForComposedType(codeInterface, deserializerMethod);
+            var deserializerFunction = new CodeFunction(method) { Name = method.Name };
+            deserializerFunction.AddUsing(deserializerMethod.Usings.ToArray());
+            children.Add(deserializerFunction);
+        }
+
+        children.Remove(deserializerMethod);
+        RemoveChildElementFromInterfaceAndNamespace(codeInterface, codeNamespace, deserializerMethod);
+    }
+
+    private static void RemoveChildElementFromInterfaceAndNamespace(CodeInterface codeInterface, CodeNamespace codeNamespace, CodeFunction function)
+    {
+        codeInterface.RemoveChildElement(function);
+        codeNamespace.RemoveChildElement(function);
+    }
+
+    private static CodeMethod CreateFactoryMethodForComposedType(CodeInterface codeInterface, CodeComposedTypeBase composedType, CodeFunction function)
+    {
+        var method = CreateCodeMethod(codeInterface, function);
+        if (composedType is not null && IsComposedOfPrimitives(composedType))
+            method.ReturnType = composedType;
+        return method;
+    }
+
+    private static CodeMethod CreateSerializerMethodForComposedType(CodeInterface codeInterface, CodeFunction function, CodeComposedTypeBase composedType)
+    {
+        var method = CreateCodeMethod(codeInterface, function);
+        // Add the key parameter if the composed type is a union of primitive values
+        if (IsComposedOfPrimitives(composedType))
+            method.AddParameter(CreateKeyParameter());
+        return method;
+    }
+
+    private static CodeMethod CreateDeserializerMethodForComposedType(CodeInterface codeInterface, CodeFunction function)
+    {
+        var method = CreateCodeMethod(codeInterface, function);
+        return method;
+    }
+
+    private static CodeMethod CreateCodeMethod(CodeInterface codeInterface, CodeFunction function)
+    {
+        var method = new CodeMethod
```

Comment:
we've implemented static copy methods instead in other cases like that to reduce the complexity in the control code.